lint: update staticcheck to 2021.1.1

stevendanna opened the pull request:
This version contains performance improvements according to the
upstream changelog.

The biggest downside is that the whole-program mode has been removed
from the unused linter. For information about why it was removed see

https://staticcheck.io/changes/2020.2/#unused-whole-program

See individual commits for fixes to new violations added by the new version.

stevendanna committed:
cli: ignore lint SA1019

This lint checks for the use of context-aware APIs that aren't
supported by DriverConn.

Release note: None

stevendanna committed:
*: fix S1025 violations

This lint checks for unnecessary uses of Sprintf. I don't love this
linter, but there were only a few violations of it.

Release note: None

stevendanna committed:
roachtest: fix SA6000 violation

This lint checks for uses of regular expressions in a loop where the
regular expresion compilation can be hoisted.

Release note: None

stevendanna committed:
geo: ignore SA4026 lint violation

Release note: None

stevendanna committed:
*: fix S1040 (unnecessary type assertion) violations

This removes type assertions on variables that are already the type
being asserted. This isn't a completely equivalent transformation,
since in the case of nil interface types, these type assertions would
have previously produced a panic.

Release note: None

stevendanna committed:
storage: fix SA4023 violation

A comparison with the untyped nil here would never be true in this
case because of the cast.

Release note: None

stevendanna committed:
sql: fix ineffectual assignment in tests

The SetNotTight function modifies the receiver so needs a pointer
receiver to be useful.

Release note: None

stevendanna committed:
util/log: fix S1017 violation

strings.TrimSuffix does a HasSuffix check, so it can be called
unconditionally.

Release note: None

stevendanna committed:
lint: update staticcheck to 2021.1.1

This version contains performance improvements according to the
upstream changelog.

The biggest downside is that the whole-program mode has been removed
from the unused linter. For information about why it was removed see

https://staticcheck.io/changes/2020.2/#unused-whole-program

This PR also updates the following additional dependencies

- [github.com/BurntSushi/toml v0.3.1...v0.4.1](https://github.com/BurntSushi/toml/compare/v0.3.1...v0.4.1)
- [golang.org/x/mod v0.4.2...v0.5.1](https://github.com/golang/mod/compare/v0.4.2...v0.5.1)
- [golang.org/x/sys 69cdffdb9359...a2f17f7b995](https://github.com/golang/sys/compare/69cdffdb9359...a2f17f7b995c)
- [golang.org/x/tools v0.1.5...v0.1.7](https://github.com/golang/tools/compare/v0.1.5...v0.1.7)

Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/72313)
<!-- Reviewable:end -->

stevendanna commented:
Not very scientific, but on this branch it ran a bit faster than the last run on master.

Master:
<img width="1221" alt="Screenshot 2021-11-02 at 10 40 00" src="https://user-images.githubusercontent.com/852371/139832332-99e4500f-d486-4b8a-8845-96902d40de34.png">

Branch:
<img width="1228" alt="Screenshot 2021-11-02 at 10 39 26" src="https://user-images.githubusercontent.com/852371/139832342-47fa645d-5483-4d8e-89d5-d3f9c99dea5b.png">

stevendanna commented:
I don't feel particularly strongly about this. I was hoping the speedup would be a bit more dramatic.

RaduBerinde reviewed (commented):
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>

Thanks for this! We should upgrade regardless - it will make it easier to stay up to date in the future.

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/72313#-:-MnZRfiGBHAjadQvCDbD:bzbus7i)* status: :shipit: complete! 1 of 0 LGTMs obtained (waiting on @dt)



<!-- Sent from Reviewable.io -->

tbg commented on pkg/geo/geomfn/angle.go:
```diff
@@ -110,7 +110,8 @@ func angleFromCoords(c1, c2, c3, c4 geom.Coord) *float64 {
 	// semantic clarity.
-	if angle == -0.0 { //lint:ignore SA4026
+	//lint:ignore SA4026 -0.0 used here for clarity
+	if angle == -0.0 {
```
Please actually fix up the other commit with this change.

stevendanna replied:
Thanks for the reminder!

tbg reviewed (approved):
LGTM

I also think this should be merged.

stevendanna committed:
spanconfig: fix ineffective assignment

This method sets the `initialized` field, so needs a pointer receiver
for the assignment to be effective.

Release note: None

stevendanna commented:
We'll need to hold off merging this for a minute. It currently likely causes TestRoachVet to fail on macOS. It appears that one of the updates we pulled in means that we now see the build warning described in https://github.com/cockroachdb/cockroach/issues/70819 as part of the roachvet output. Unfortunately, that means that RoachVet fails since it assumes any output is a failure.

stevendanna commented:
https://github.com/cockroachdb/cockroach/pull/72686 should resolve the warning

srosenberg reviewed (commented):
> Reviewed 2 of 2 files at r1, 34 of 34 files at r21, 3 of 3 files at r22, 1 of 1 files at r23, 1 of 1 files at r24, 20 of 20 files at r25, 1 of 1 files at r26, 1 of 1 files at r27, 1 of 1 files at r28.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/72313)* status: :shipit: complete! 0 of 0 LGTMs obtained (and 1 stale) (waiting on @dt, @srosenberg, @stevendanna, and @tbg)

___
*[pkg/spanconfig/spanconfigkvsubscriber/kvsubscriber.go, line 423 at r29](https://reviewable.io/reviews/cockroachdb/cockroach/72313#-MoJqVeR0TP6U0I6sQfF:-MoJqVeR0TP6U0I6sQfG:by3qruz) ([raw file](https://github.com/cockroachdb/cockroach/blob/854ed53ddf4cb9dbcd8d70a258c8987893bc77b7/pkg/spanconfig/spanconfigkvsubscriber/kvsubscriber.go#L423)):*
> ```Go
> }
> 
> func (h *handler) invoke(update roachpb.Span) {
> ```

Another bug...none of the below updates to `h` were previously persisted, so why wasn't anything downstream affected?
___
*[pkg/sql/inverted/expression_test.go, line 84 at r27](https://reviewable.io/reviews/cockroachdb/cockroach/72313#-MoJpw8cDgAO_R6rZ3S-:-MoJpw8cDgAO_R6rZ3S0:b28at99) ([raw file](https://github.com/cockroachdb/cockroach/blob/196f6b37e694ca45af958b64a03590f2b0de5b4f/pkg/sql/inverted/expression_test.go#L84)):*
> ```Go
> 
> func (u *UnknownExpression) IsTight() bool { return u.tight }
> func (u *UnknownExpression) SetNotTight()  { u.tight = false }
> ```

Surprised nothing else was broken considering the previous setter was effectively no-op?


<!-- Sent from Reviewable.io -->

srosenberg reviewed (approved):
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/72313#-:-MoJqzHN64FWQseXCUU8:bnfp4nl)* status: :shipit: complete! 1 of 0 LGTMs obtained (and 1 stale) (waiting on @dt, @srosenberg, @stevendanna, and @tbg)



<!-- Sent from Reviewable.io -->

stevendanna commented:
bors r=srosenberg,RaduBerinde,tbg
